src/iterated_integrals.jl

```diff
@@ -0,0 +1,237 @@
+    WikJ::WikJType
+end
+
+mutable struct WikJCommute_iip{WikJType} <: AbstractWikJCommute
+    WikJ::WikJType
+end
+
+struct WikJGeneral_oop{rateNoiseElTypeNoUnits, WikJType} <: AbstractWikJGeneral
+    WikJ::WikJType
+    m_seq::Array{Int}
+end
+
+mutable struct WikJGeneral_iip{rateNoiseElTypeNoUnits, WikJType} <: AbstractWikJGeneral
+    WikJ::WikJType
+    WikJ2::WikJType
+    WikJ3::WikJType
+    m_seq::Array{Int}
+    vec_ζ::Vector{eltype(rateNoiseElTypeNoUnits)}
+    vec_η::Vector{eltype(rateNoiseElTypeNoUnits)}
+    Gp1::Vector{eltype(rateNoiseElTypeNoUnits)}
+    Gp2::Vector{eltype(rateNoiseElTypeNoUnits)}
+end
+
+function fill_WikJDiagonal_oop(ΔW)
+    WikJ = false .* ΔW .* ΔW
+    WikJDiagonal_oop{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJDiagonal_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW
+    WikJDiagonal_iip{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJCommute_oop(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_oop{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJCommute_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_iip{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJGeneral_oop(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    WikJCommute_oop{eltype(ΔW), typeof(WikJ)}(WikJ, m_seq)
+end
+
+function fill_WikJGeneral_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJ2 = similar(WikJ)
+    WikJ3 = similar(WikJ)
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    vec_ζ = false .* vec(ΔW)
+    vec_η = similar(vec_ζ)
+    Gp1 = zeros(M)
+    Gp2 = similar(Gp1)
+    WikJCommute_iip{eltype(ΔW), typeof(WikJ)}(WikJ, WikJ2, WikJ3, m_seq, vec_ζ, vec_η, Gp1, Gp2)
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_oop)
+    @unpack WikJ = Wik
+    WikJ = 1//2 .* dW .* dW
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_iip)
+    @unpack WikJ = Wik
+    if typeof(dW) <: Number
+        Wik.WikJ = 1//2 .* dW .^ 2
+    else
+        @.. WikJ = 1//2*dW^2
+        @.. Wik.WikJ = WikJ
+    end
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_oop)
+    @unpack WikJ = Wik
+    WikJ = 1//2 .* vec(dW) .* vec(dW)'
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_iip)
+    @unpack WikJ = Wik
+    mul!(WikJ,vec(dW),vec(dW)')
+    @.. WikJ *= 1//2
+
+    @.. Wik.WikJ = WikJ
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJGeneral_oop)
+    @unpack WikJ, m_seq = Wik
+
+    m      = length(dW)
+    M      = m*(m-1)/2
+    sum_dW² = dW'*dW
+
+    WikJ = dW*dW'
+    Gp1 = randn(M)
+    α = sqrt(1 + sum_dW²/dt)
+    Gp2 = Gp1/(sqrt(2)*(1+α)*dt)
```

Comment:
I fail to see how this is Sigma infty?

Reply:
No. Actually, I've changed the order in which we multiply scalars. Sigma infty is `a_p`

Reply:
can you rename a_p then?

Reply:
Oh sorry, you are referring to the operator sigma infinity. I broke that down into some smaller operators.